homeassistant/components/modbus/modbus.py

```diff
@@ -202,6 +213,41 @@ async def async_setup(self):
         async with self._lock:
             await self.hass.async_add_executor_job(self._pymodbus_connect)
 
+        self._call_type = {
```

Comment:
This can be defined in init. Alternatively move it to the module level by replacing the methods with the names of the methods and use `getattr` to get the corresponding client method before calling it.

Reply:
That is an alternative solution but I read that getattr is expensive. How about I declare it in init and complete the function in setup() ?

Reply:
I have made a compromise, which I hope you find acceptable. Declaring the variable in init is the right way, but using `getattr`to select the pymodbus function is putting a lot of unneeded load on the system. Please remember that modbus typically have many sensors etc. configured, one of the users have around 200 sensor and a scan_interval of 5, it runs on a raspberry pi.

The declaration can be in __init__ but the client part of the function call is not available. I can of course merge setup with __init__, but I really do not like to have real functionality in a constructor.

I am trying to keep making the performance (cpu load) better, and all places where I have read about `getattr`earlier tell it is an expensive function.

Of course if you insist, I will without further comments implement the `getattr`. Please advice.